mithril-aggregator/src/http_server/routes/proof_routes.rs

```diff
@@ -70,6 +76,14 @@ mod handlers {
             transaction_parameters.transaction_hashes
         );
 
+        if let Err(error) = validator.validate(&transaction_hashes) {
+            warn!("proof_cardano_transaction::bad_request");
+            return Ok(reply::bad_request(error.label, error.message));
+        }
+
+        transaction_hashes.sort();
+        transaction_hashes.dedup();
```

Comment:
IMO this could be done by the validator

Reply:
We've moved the responsibility to the `CardanoTransactionProofQueryParams` struct. Commit: https://github.com/input-output-hk/mithril/pull/1774/commits/7a9e6ea17c70f2935867749cec8e240d6133ec3c